demos/sample_azure_iot/sample_azure_iot.c

```diff
@@ -0,0 +1,624 @@
+
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/* Standard includes. */
+#include <string.h>
+#include <stdio.h>
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* Demo Specific configs. */
+#include "demo_config.h"
+
+/* Azure Provisioning/IoTHub library includes */
+#include "azure_iot_hub_client.h"
+#include "azure_iot_provisioning_client.h"
+
+/* Exponential backoff retry include. */
+#include "backoff_algorithm.h"
+
+/* Transport interface implementation include header for TLS. */
+#include "transport_tls_socket.h"
+
+/* Crypto helper header. */
+#include "crypto.h"
+/*-----------------------------------------------------------*/
+
+/* Compile time error for undefined configs. */
+#if !defined( democonfigHOSTNAME ) && !defined( democonfigENABLE_DPS_SAMPLE )
+    #error "Define the config democonfigHOSTNAME by following the instructions in file demo_config.h."
+#endif
+
+#if !defined( democonfigENDPOINT ) && defined( democonfigENABLE_DPS_SAMPLE )
+    #error "Define the config dps endpoint by following the instructions in file demo_config.h."
+#endif
+
+#ifndef democonfigROOT_CA_PEM
+    #error "Please define Root CA certificate of the MQTT broker(democonfigROOT_CA_PEM) in demo_config.h."
+#endif
+
+#if defined( democonfigDEVICE_SYMMETRIC_KEY ) && defined( democonfigCLIENT_CERTIFICATE_PEM )
+    #error "Please define only one auth democonfigDEVICE_SYMMETRIC_KEY or democonfigCLIENT_CERTIFICATE_PEM in demo_config.h."
+#endif
+
+#if !defined( democonfigDEVICE_SYMMETRIC_KEY ) && !defined( democonfigCLIENT_CERTIFICATE_PEM )
+    #error "Please define one auth democonfigDEVICE_SYMMETRIC_KEY or democonfigCLIENT_CERTIFICATE_PEM in demo_config.h."
+#endif
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The maximum number of retries for network operation with server.
+ */
+#define sampleazureiotRETRY_MAX_ATTEMPTS                      ( 5U )
+
+/**
+ * @brief The maximum back-off delay (in milliseconds) for retrying failed operation
+ *  with server.
+ */
+#define sampleazureiotRETRY_MAX_BACKOFF_DELAY_MS              ( 5000U )
+
+/**
+ * @brief The base back-off delay (in milliseconds) to use for network operation retry
+ * attempts.
+ */
+#define sampleazureiotRETRY_BACKOFF_BASE_MS                   ( 500U )
+
+/**
+ * @brief Timeout for receiving CONNACK packet in milliseconds.
+ */
+#define sampleazureiotCONNACK_RECV_TIMEOUT_MS                 ( 1000U )
```

Comment:
make this 10 like we have in the e2e?
```suggestion
#define sampleazureiotCONNACK_RECV_TIMEOUT_MS                 ( 10 * 1000U )
```